crates/bevy_macro_utils/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+extern crate proc_macro;
+
+use cargo_manifest::{DepsSet, Manifest};
+use proc_macro::TokenStream;
+use std::{env, path::PathBuf};
+
+pub fn get_module_path(name: &str) -> syn::Path {
+    const BEVY: &str = "bevy";
+    const BEVY_INTERNAL: &str = "bevy_internal";
+
+    let find_in_deps = |deps: DepsSet| -> Option<syn::Path> {
+        let package = if let Some(dep) = deps.get(BEVY) {
+            Some(dep.package().unwrap_or(BEVY))
+        } else if let Some(dep) = deps.get(BEVY_INTERNAL) {
+            Some(dep.package().unwrap_or(BEVY_INTERNAL))
+        } else {
+            None
+        };
+
+        package.map(get_path).map(|mut p| {
+            if let Some(module) = name.strip_prefix("bevy_") {
+                p.segments.push(parse_path(module));
+            }
+            p
+        })
+    };
+
+    let manifest = env::var_os("CARGO_MANIFEST_DIR")
+        .map(PathBuf::from)
+        .map(|mut path| {
+            path.push("Cargo.toml");
+            Manifest::from_path(path).unwrap()
+        })
+        .unwrap();
+    let deps = manifest.dependencies;
+    let deps_dev = manifest.dev_dependencies;
+
+    manifest
+        .package
+        .and_then(|p| {
+            if p.name == name && env::var("CARGO_PRIMARY_PACKAGE").is_ok() {
```

Comment:
The second part of the condition fixes issues with doc examples, so they can be built and run, but unfortunately clippy even being the compiler based, fails with numerous errors on `bevy_reflect`. Committed as is since ran out of ideas, but will open an issue in clippy's repo and see what else can be done.

Reply:
`CARGO_PRIMARY_PACKAGE` won't be set when using bevy as dependency, so I think it is wrong to use it here.